src/Utility/TextSplitter.cpp

```diff
@@ -1,35 +1,35 @@
 #include "TextSplitter.h"
 
-size_t TextSplitter::numOfWindows(size_t textLength, int patternLength) {
+size_t TextSplitter::numOfWindows(size_t textLength, size_t patternLength) {
     return textLength - patternLength + 1;
 }
 
 std::pair<size_t, size_t> TextSplitter::lenOfChains(size_t numOfWindows) {
-    int rangeWidth = numOfWindows / numOfThreads;
-    int rangesWithBonus = numOfWindows % numOfThreads;
+    auto rangeWidth = numOfWindows / numOfThreads;
+    auto rangesWithBonus = numOfWindows % numOfThreads;
```

Comment:
Same question here :smile:

Reply:
Here if you type int you are doing a conversion from unsigned to signed datatype. Auto will resolve in this case to size_t.